src/d/semantic/caster.d

```diff
@@ -650,3 +676,96 @@ struct Caster(bool isExplicit, alias bailoutOverride = null) {
 		return CastKind.Invalid;
 	}
 }
+
+@("Test implicit casts of floats to/from BasicTypes")
+unittest {
+	import source.location;
+	with (BuiltinType) {
+		void assertCastKind(const CastKind ck, const CastKind shouldBe) {
+			if (ck != shouldBe) {
+				import std.format;
+				assert(
+					0,
+					format("Cast yielded `%s`, when `%s` was expected", ck,
+					       shouldBe)
+				);
+			}
+		}
+
+		BuiltinType[] floatTypes = [Float, Double, Real];
+
+		// bool to float
+		foreach (floatType; floatTypes) {
+			Type from = Type.get(Bool);
+			Type to = Type.get(floatType);
+
+			const implicit = implicitCastFrom(null, from, to);
```

Comment:
`implicit` is a strange name here.

This whole loop could be a single test on its own. This i another strong indicator that case from bool is its own beast. All the code is separated, and in the only instance it's not in this PR, the code would have been simpler if it was.

Reply:
gone